nodejs/week3/calculatorRouter.js

```diff
@@ -0,0 +1,70 @@
+
+const express = require("express");
+const calculatorRouter = express.Router();
+
+// /calculator/addition?firstParam=1&secondParam=2 should respond with 3.
+
+calculatorRouter.get('/addition', (request,response) => {
+
+  console.log(request.query);
+  if (request.query.firstParam === {num1}, request.query.secondParam === {num2} ) { 
+
+  // New app using express module 
+  const num1 = parseInt(request.params.num1);
+  const num2 = parseInt(request.params.num2);
+  
+  var result =  num1 + num2;
+  
+  response.send(result); 
+};});
+
+// /calculator/subtration?firstParam=4&secondParam=2 should respond with 2.
+
+calculatorRouter.get('/subtration', (request,response) => {
+  console.log(request.query);
+  if (request.query.firstParam === {num1}, request.query.secondParam === {num2} ) { 
+
+  // New app using express module 
+
+  const num1 = parseInt(request.params.num1);
+  const num2 = parseInt(request.params.num2);
+  
+  var result =  num1 - num2;
+  
+  response.send(result); 
+};});
+
+// /calculator/multiply?firstParam=4&secondParam=2 should respond with 8.
+
+calculatorRouter.get('/multiply', (request,response) => {
+  console.log(request.query);
+  if (request.query.firstParam === {num1}, request.query.secondParam === {num2} ) { 
+
+
+  // New app using express module 
+  
+  const num1 = parseInt(request.params.num1);
+  const num2 = parseInt(request.params.num2);
+  
+  var result =  num1 * num2;
+  
+  response.send(result); 
+};});
+
+// /calculator/multiply?firstParam=8&secondParam=2 should respond with 4.
+
+calculatorRouter.get('/Division', (request,response) => {
+    console.log(request.query);
+
+    if (request.query.firstParam === {num1}, request.query.secondParam === {num2} ) { 
+
+    // New app using express module 
+    const num1 = parseInt(request.params.num1);
+    const num2 = parseInt(request.params.num2);
+    
+    var result =  num1 / num2;
```

Comment:
My comment from above also applies here. What is the purpose of defining the variable `result`? Doing the following would be the same:

    response.send(parseInt(request.params.num1) / parseInt(request.params.num2);

This does everything needed in a single line, and I find it easier to read. Unless you use a variable that you define more than once or do some transformation on it, I prefer not defining the variable. But this is an opinion. Of course when you do this:

    const num1 = parseInt(request.params.num1);

You are doing a transformation on the original value. Just wanted to give another example of how it can be written. Sometimes being verbose is useful, but other times it is unnecessary.